Bleak Fork Upstream Update

pmessan opened the pull request:
This PR proposes an update to use the proglove_next branch as our updated bleak fork. A few patches were added including resume and suspend methods to the BleakScannerBlueZDBus class, and setup for Cloudsmith.

Olivier-ProGlove committed:
clean(packaging): Rename python package as 'bleak-proglove'

This package renaming is to avoid dependency issue with the official 'bleak' python package.
This fork should be refered as 'bleak-proglove'.

The python package is still named 'bleak' in the source code.

Olivier-ProGlove committed:
ci: Package 'bleak_proglove' and publish it on Cloudsmith

arthur-proglove committed:
feat(version): Append a development number to the version

Append a `.dev.N` to the version number which matches the number of
commit from HEAD to the last commit of the `__version.py` file.
A number of commits equal to 25 will produce the following version:
    x.y.z.dev25
A number of commits equal to zero will produce the following version:
    x.y.z

arthur-proglove committed:
feat(package): Use post instead of dev for in between releases versions

Pip will use this version as the latest

arthur-proglove committed:
chore(build): Use new Cloudsmith repository

pmessan committed:
fix(scanner): Add suspend and resume to scanner

pmessan committed:
fix(build): Update build branch

arthur-proglove commented on bleak/backends/bluezdbus/scanner.py:
```diff
@@ -141,53 +141,69 @@ async def start(self):
-        # Apply the filters
+        await self.resume()
+
+    async def stop(self):
```
why did you move it here? It makes the diff impossible to understand.

pmessan replied:
Reorganised it now

arthur-proglove commented on bleak/backends/bluezdbus/scanner.py:
```diff
@@ -141,53 +141,69 @@ async def start(self):
+
+        self._bus = None
+
+    # add suspend()
```
what is that comment?

pmessan replied:
Removed

arthur-proglove commented on bleak/backends/bluezdbus/scanner.py:
```diff
@@ -141,53 +141,69 @@ async def start(self):
-        # Start scanning
+    # add resume()
+    async def resume(self):
+        # if self.is_scanning:
+        #     return
```
what is that dead code?

pmessan replied:
Removed

arthur-proglove commented on bleak/backends/bluezdbus/scanner.py:
```diff
@@ -141,53 +141,69 @@ async def start(self):
-                interface=defs.ADAPTER_INTERFACE,
-                member="StartDiscovery",
-            )
+        Message(
+            destination=defs.BLUEZ_SERVICE,
+            path=self._adapter_path,
+            interface=defs.ADAPTER_INTERFACE,
+            member="SetDiscoveryFilter",
+            signature="a{sv}",
+            body=[self._filters],
+                )
```
why did the indentation change?

pmessan replied:
Fixed

arthur-proglove commented on bleak/backends/bluezdbus/scanner.py:
```diff
@@ -141,53 +141,69 @@ async def start(self):
             )
         )
         assert_reply(reply)
+        # except RemoteError as e:
+        #     logger.error("Stop discovery failed: {0}".format(e))
```
what is that dead code?

pmessan replied:
Removed

arthur-proglove commented on bleak/backends/bluezdbus/scanner.py:
```diff
@@ -141,53 +141,69 @@ async def start(self):
-            self._bus.disconnect()
-        except Exception as e:
-            logger.error("Attempt to disconnect system bus failed: {0}".format(e))
+        # self.is_scanning = True
```
why was this added?

pmessan replied:
Removed

arthur-proglove reviewed (changes requested):
Are you sure you added all the important commits? Only 2 looks very small to me.

- From what I see there are at least 3 commits on `setup.py` that should make it here.
- Do we have a new method to get the MTU in the upstream bleak? (1f06df9171eb99e55a1a9c0b05d74c9533a9f92e)
- Do we still support the skip discovery flag? (2115ef0079bd07018a45a144971e65667e53eae2)
- What about the cache? (8e9678a10d0245e6dbe11df07ad56ab4ad21d0cd, 93e783f1b16f91fb822724138c90203dd10bf74b, 43ecf105d838342e93215cb065bc9733dcbe476f)
- Do we still check for BlueZ crashses (cad63b0da22bf65b2a81435bc33196fe92e2d8c1) ?

pmessan committed:
fix(client): Check if caching is enabled

pmessan commented on bleak/backends/bluezdbus/scanner.py:
```diff
@@ -189,6 +196,43 @@ async def stop(self):
+            )
+        )
+    async def _restart(self):
+        """Stop and restart the discovery."""
```
I'm a little lost as to where this is called exactly.

pmessan commented:
> Are you sure you added all the important commits? Only 2 looks very small to me.
> 
> * From what I see there are at least 3 commits on `setup.py` that should make it here.

Looked at it, and added a commit with the releva.nt changes

> * Do we have a new method to get the MTU in the upstream bleak? ([1f06df9](https://github.com/workaroundgmbh/bleak/commit/1f06df9171eb99e55a1a9c0b05d74c9533a9f92e))

In the upstream branch, There's already a [method ](https://github.com/workaroundgmbh/bleak/blob/d6c208f15bf3ad84338d6904dc0a2ee1242e27a9/bleak/backends/bluezdbus/client.py#L519)called `_acquire_mtu` which does this.
 
> * Do we still support the skip discovery flag? ([2115ef0](https://github.com/workaroundgmbh/bleak/commit/2115ef0079bd07018a45a144971e65667e53eae2))

I think we don't need to any more because the upstream `get_services` [method](https://github.com/workaroundgmbh/bleak/blob/d6c208f15bf3ad84338d6904dc0a2ee1242e27a9/bleak/backends/bluezdbus/client.py#L585) only returns the services if they are resolved, hence fixing the problem of bluez crashing if we don't wait for the services to be resolved.

> * What about the cache? ([8e9678a](https://github.com/workaroundgmbh/bleak/commit/8e9678a10d0245e6dbe11df07ad56ab4ad21d0cd), [93e783f](https://github.com/workaroundgmbh/bleak/commit/93e783f1b16f91fb822724138c90203dd10bf74b), [43ecf10](https://github.com/workaroundgmbh/bleak/commit/43ecf105d838342e93215cb065bc9733dcbe476f))

I added the patches for the cache, since I did not find any pre-exisiting code that checked if the cache was enabled. I think those patches are still relevant.

> * Do we still check for BlueZ crashses ([cad63b0](https://github.com/workaroundgmbh/bleak/commit/cad63b0da22bf65b2a81435bc33196fe92e2d8c1)) ?

Agreed to not focus on this for now

pmessan commented:
> > Are you sure you added all the important commits? Only 2 looks very small to me.
> > 
> > * From what I see there are at least 3 commits on `setup.py` that should make it here.
> > * Do we have a new method to get the MTU in the upstream bleak? ([1f06df9](https://github.com/workaroundgmbh/bleak/commit/1f06df9171eb99e55a1a9c0b05d74c9533a9f92e))
> 
> In the upstream branch, There's already a method called `_acquire_mtu` which does this.
> 
> > * Do we still support the skip discovery flag? ([2115ef0](https://github.com/workaroundgmbh/bleak/commit/2115ef0079bd07018a45a144971e65667e53eae2))
> 
> I think we don't need to any more because the upstream `get_services` [method](https://github.com/workaroundgmbh/bleak/blob/d6c208f15bf3ad84338d6904dc0a2ee1242e27a9/bleak/backends/bluezdbus/client.py#L585) only returns the services if they are resolved, hence fixing the problem of bluez crashing if we don't wait for the services to be resolved.
> 
> > * What about the cache? ([8e9678a](https://github.com/workaroundgmbh/bleak/commit/8e9678a10d0245e6dbe11df07ad56ab4ad21d0cd), [93e783f](https://github.com/workaroundgmbh/bleak/commit/93e783f1b16f91fb822724138c90203dd10bf74b), [43ecf10](https://github.com/workaroundgmbh/bleak/commit/43ecf105d838342e93215cb065bc9733dcbe476f))
> 
> I added the patches for the cache, since I did not find any pre-exisiting code that checked if the cache was enabled. I think those patches are still relevant.
> 
> > * Do we still check for BlueZ crashses ([cad63b0](https://github.com/workaroundgmbh/bleak/commit/cad63b0da22bf65b2a81435bc33196fe92e2d8c1)) ?

Agreed to not focus on this for now

arthur-proglove commented on bleak/backends/bluezdbus/client.py:
```diff
@@ -84,6 +87,28 @@ def __init__(self, address_or_ble_device: Union[BLEDevice, str], **kwargs):
 
+
+    def _is_cache_enabled() -> bool:
+        global _cache_enabled
```
this could be a class member

pmessan replied:
Fixed

arthur-proglove commented on bleak/backends/bluezdbus/scanner.py:
```diff
@@ -143,6 +144,42 @@ async def start(self):
 
         await self.resume()
 
+    async def resume(self):
```
This does not really match the commit description?

pmessan replied:
I think it does though, it was added in [this commit](https://github.com/workaroundgmbh/bleak/pull/33/commits/61c12e54b5750bae751d78889c1bfb0d5d953497#diff-cc469ec95c55bf20bb32d39bf09be9cec2dfc6820e40e857367f8b49915cddcdR175)

arthur-proglove reviewed (changes requested):
You did not cherry pick the commits but created your own for some changes that could just have been taken from the git history (setup.py and .travis.yml). I updated this PR with the proper commits.

arthur-proglove commented on bleak/backends/bluezdbus/scanner.py:
```diff
@@ -189,6 +196,43 @@ async def stop(self):
+                body=[defs.ADAPTER_INTERFACE, 'Powered', Variant("b", True)],
+            )
+        )
+    async def _restart(self):
```
this is not even called

pmessan replied:
removed

arthur-proglove commented on bleak/backends/bluezdbus/scanner.py:
```diff
@@ -165,7 +169,7 @@ async def start(self):
         assert_reply(reply)
 
-    async def stop(self):
+    async def suspend(self):
```
so `stop` just became `suspend`? :-D

pmessan replied:
No, `suspend` is called inside of stop, like how it was done [here](https://github.com/workaroundgmbh/bleak/blob/859f7b966bbabdff29908e07e534f09e8b49ee40/bleak/backends/bluezdbus/discovery.py#L256)

arthur-proglove commented:
Okay looks good to me, we're just missing `_acquire_mtu` to be exposed. Somehow there is a property `mtu_size` but since `_acquire_mtu` is not called anywhere `_mtu_size` is not set. 
I think `_acquire_mtu` should be called `once` after being connected, maybe add one more commit with this patch?

pmessan committed:
fix(client): Call `_acquire_mtu` method in `connect`

pmessan commented:
> Okay looks good to me, we're just missing `_acquire_mtu` to be exposed. Somehow there is a property `mtu_size` but since `_acquire_mtu` is not called anywhere `_mtu_size` is not set. I think `_acquire_mtu` should be called `once` after being connected, maybe add one more commit with this patch?

Patch added!

arthur-proglove commented on bleak/backends/bluezdbus/client.py:
```diff
@@ -340,6 +340,8 @@ async def connect(self, **kwargs) -> bool:
             # Get all services. This means making the actual connection.
             await self.get_services()
 
+            await self._acquire_mtu()
```
you're not exposing it ;-) you're calling it

pmessan replied:
Reworded the commit 😆

arthur-proglove committed:
feat(travis): Use Python 3.7

arthur-proglove reviewed (approved):
LGTM :-)

pmessan commented:
Apparently I'm not authorized to merge 😆 
How do I fix that?

arthur-proglove commented:
> Apparently I'm not authorized to merge laughing How do I fix that?

let me check
